src/commons/utils/InfiniteLoopReporter.ts

```diff
@@ -0,0 +1,162 @@
+import * as Sentry from '@sentry/browser';
+import { Context, NativeStorage, SourceError } from 'js-slang/dist/types';
+
+/**
+ * Parses the error and determines whether it is an infinite loop and
+ * if so, what kind of infinite loop.
+ *
+ * @returns {string} string containing the infinite loop classification.
+ * @returns {string} empty string, if the error is not an infinite loop.
+ */
+function infiniteLoopErrorType(errorString: string): string {
+  if (errorString.includes('Infinite recursion (or runtime error) detected')) {
+    if (errorString.includes('forget your base case')) return 'no_base_case';
+    else if (
+      errorString.includes('Did you call a value that is outside the range of your function?')
+    )
+      return 'input_out_of_domain';
+    else if (errorString.includes('Check your recursive function calls.')) return 'no_state_change';
+  }
+  // firefox
+  else if (errorString === 'InternalError: too much recursion') return 'stack_overflow';
+  // chrome
+  else if (errorString === 'RangeError: Maximum call stack size exceeded') return 'stack_overflow';
+  // edge
+  else if (errorString === 'Error: Out of stack space') return 'stack_overflow';
```

Comment:
I'm a bit wary of hardcoding this since they could very well change between browser versions (and Safari is missing here, of the few popular browsers), but I'm not sure if there's any better way..

Reply:
I did some testing and it turns out Safari's error message is the same as Chrome's. (added this info as a comment in the code). Also moved this bit to an enum in js-slang